woocommerce/woocommerce
# Feature/css custom props - followup #29575

Originally created by @helgatheviking #29575

### All Submissions:

Closes https://github.com/woocommerce/woocommerce/issues/29574 .

How to test the changes in this Pull Request:
- Rebuild the CSS files 

### Changes proposed in this Pull Request:
- Replace SCSS color variables with CSS color variables. This will allow a theme to override the CSS root variable and have that change propagate to all the places where that color variable is used. Currently, overriding the root variable does nothing as the colors are still hard-coded into the CSS when built from the SCSS.

### How to test the changes in this Pull Request:
1. Check the inlined style in header (woocommerce-inline-inline-css), has to contain the css custom props
2. Check the compiled css style and search for /(#[a-fA-F0-9]{9}|#[a-fA-F0-9]{7}|#[a-fA-F0-9]{4}|rgb(.*\))|rgba(.*\)))/ to check that all colors have been converted
3. With devtools change the --wc--primary-hue to another value, button and highlight will change.  do the same in admin section and nothing happens (the reason is because in the wp-admin section the --wc-orchid color was used)
4. added a filter "enqueue_woo_global_styles" in to let the user to modify/add/remove the woo-commerce inline style. 

### FOR PR REVIEWER ONLY:
- I used this... file_get_contents, I don't know if this can create any security issue.
